### ISSUE TITLE
AMD Radeon RX 5600 XT / gfx1010 timings are not supported

### ISSUE THREAD

**Issue by wtu-fl** [nanopool/nanominer#69]
Trying out the new AMD Radeon RX 5600 XT (Sapphire Pulse brand) after reading AMAZING things about it's power efficiency for mining Ethereum, in PC World review.  However, Nanominer 1.8.2 doesn't like it.  Log says:
<error> Device #0 (architecture: gfx1010) timings are not supported.
...right before it quits.  What a shame.  Hope you can update for this soon?
Otherwise, love your software...  thanks!

**Comment by Grumpy-Dwarf** [comment]
Timings update in runtime is not yet supported for those GPUs but mining is fully supported since 1.7 version. Please check you have latest nanominer version. If it still quits, please provide the full log for investigation.

**Comment by wtu-fl** [comment]
Here is the Log:  GPU0 is the AMD Radeon RX 5600 XT, GPUs 1 thru 5 are Nvidea.

Also tried with config.ini devices=0, to try the AMD alone, but had the exact same results.

Note, the miner worked fine with config .ini devices=1,2,3,4,5 

(Windows 7 Pro 64 bit)

 

2020-02-20 14:45:14: <info>                                    _                 

2020-02-20 14:45:14: <info>  _ __   __ _ _ __   ___  _ __ ___ (_)_ __   ___ _ __ 

2020-02-20 14:45:14: <info> | '_ \ / _` | '_ \ / _ \| '_ ` _ \| | '_ \ / _ \ '__|

2020-02-20 14:45:14: <info> | | | | (_| | | | | (_) | | | | | | | | | |  __/ |   

2020-02-20 14:45:14: <info> |_| |_|\__,_|_| |_|\___/|_| |_| |_|_|_| |_|\___|_|   

2020-02-20 14:45:14: <info> 

2020-02-20 14:45:14: <info> Version 1.8.2

2020-02-20 14:45:14: <info>  Started EthMan server on port 3333 without a password (read-only mode).

2020-02-20 14:45:14: <info>  Started web interface on port 9090 without a password.

2020-02-20 14:45:14: <info>  CUDA driver version is 10.2, runtime version is 10.0

2020-02-20 14:45:16: <info> Detected 6 devices

2020-02-20 14:45:16: <info> GPU 0 PCI 03:00.0, Platform: OpenCL, Name: AMD Radeon RX 5600 XT (gfx1010), 6128 MB available

2020-02-20 14:45:16: <info> GPU 1 PCI 08:00.0, Platform: CUDA, Name: GeForce GTX 1070, 8192 MB available

2020-02-20 14:45:16: <info> GPU 2 PCI 09:00.0, Platform: CUDA, Name: GeForce GTX 1070, 8192 MB available

2020-02-20 14:45:16: <info> GPU 3 PCI 0d:00.0, Platform: CUDA, Name: GeForce GTX 1070, 8192 MB available

2020-02-20 14:45:16: <info> GPU 4 PCI 0f:00.0, Platform: CUDA, Name: GeForce GTX 1070, 8192 MB available

2020-02-20 14:45:16: <info> GPU 5 PCI 10:00.0, Platform: CUDA, Name: GeForce GTX 1070, 8192 MB available

2020-02-20 14:45:16: <info> -------------------Ethash:------------------

2020-02-20 14:45:16: <info> Currency: Ethereum

2020-02-20 14:45:16: <info> Wallet: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

2020-02-20 14:45:16: <info> Rig name: Sneezy_5ade92a5

2020-02-20 14:45:16: <info> Email was not set.

2020-02-20 14:45:16: <info> Hashrate threshold to restart miner is not set

2020-02-20 14:45:16: <info> Added pool: eth-us-east1.nanopool.org:9433

2020-02-20 14:45:16: <info> Added pool: eth-us-west1.nanopool.org:9433

2020-02-20 14:45:16: <info> Added pool: eth-eu2.nanopool.org:9433

2020-02-20 14:45:16: <info> Added pool: eth-eu1.nanopool.org:9433

2020-02-20 14:45:16: <info> Added pool: eth-jp1.nanopool.org:9433

2020-02-20 14:45:16: <info> Added pool: eth-au1.nanopool.org:9433

2020-02-20 14:45:16: <info> Added pool: eth-asia1.nanopool.org:9433

2020-02-20 14:45:16: <info> Auto pool sorting is enabled

2020-02-20 14:45:16: <info> Devices: 0

2020-02-20 14:45:16: <info> Memory tweak: 1,1,1,1,1,1

2020-02-20 14:45:16: <info> -------------Common parameters:-------------

2020-02-20 14:45:16: <info> Log path: ./logs/

2020-02-20 14:45:16: <info> Log is written to file

2020-02-20 14:45:16: <info> GPU watchdog is enabled

2020-02-20 14:45:16: <info> Never calling reboot.bat (always restarting the miner in case of errors). 

2020-02-20 14:45:19: <info>  Setting up algorithm: Ethash

2020-02-20 14:45:20: <info>  Connecting to pools...

2020-02-20 14:45:20: <info>  Connected to pool: eth-us-east1.nanopool.org:9433 | ping: 231 ms.

2020-02-20 14:45:21: <info>  New job from eth-us-east1.nanopool.org:9433 | share difficulty = 10000 MH, epoch 317 (3559 MB)

2020-02-20 14:45:21: <error>  Device #0 (architecture: gfx1010) timings are not supported.

2020-02-20 14:45:21: [Statistics] Ethereum - Total speed: 0.000 H/s, Total shares: 0 Rejected: 0, Time: 00:01

 

At this point, there was a Windows popup box stating that the program stopped functioning, reason unknown.

**Comment by ayushmaang** [comment]
Was there any update to this?

**Comment by wtu-fl** [comment]
Grumpy has acknowledged it is a bug.  We are still awaiting his release of a fix for this.

**Comment by tmagomedov** [comment]
>At this point, there was a Windows popup box stating that the program stopped functioning, reason unknown.

Still could not reproduce the issue. Technically nothing changed in miner, just binaries for Navi GPUs were added. This is probably AMD driver related issue.

As for unsupported timings - this warning can be safely ignored.

**Comment by wtu-fl** [comment]
Would you like me to ship you the hardware so you can reproduce the problem?  (It isn’t doing me any good right now.)

**Comment by tmagomedov** [comment]
Thanks, it has same kernel binaries as RX 5700 series which I've tested. Have you seen issue #64 ? What if you start nanominer with only AMD devices enabled in config?
Option to add for AMD-only mining on you rig:
`devices = 0`
Nvidia-only:
`devices = 1,2,3,4,5`

If it works for both cases but fails for AMD+Nvidia - we are making some progress with this issue investigation at least.

**Comment by wtu-fl** [comment]
If you study the config file I posted above, you will see that it was with devices = 0

And it STILL didn’t work.

**Comment by lmnope1** [comment]
Mining works but I get a similar error (warning) message using a 5700. Overclocked I get 49MH/s or so. GPU temperate is not being read at all though, it stays at 0!

**Comment by wtu-fl** [comment]
Really?   Since mine just bombs out, period, I’m willing to experiment.  Please let us know your Operating System version, and what AMD driver version you are using, and where you got it.  Thanks!

**Comment by lmnope1** [comment]
> Really? Since mine just bombs out, period, I’m willing to experiment. Please let us know your Operating System version, and what AMD driver version you are using, and where you got it. Thanks!

2015 MacBook Pro with AMD 5700 in an eGPU Sonnet 350 box over Thunderbolt 2. Running latest version of Windows insider build (Fast Ring April 15th). Newest beta Adrenaline drivers from AMD as of today

**Comment by Yovaniteo** [comment]
Gradias

**Comment by Yovaniteo** [comment]
Insecto

**Comment by wtu-fl** [comment]
Follow-up on original issue:  After a 2nd RMA of this Sapphire RX 5600 XT, I finally got one that works.  At its highest power efficiency, it mines at about 36 Mh/s using 87 watts.  The only issues are:
1) Nanominer doesn't report the GPU temperatures (e.g. http://localhost:9090/#/rig temperatures reported at 0; this is true of the 5700 XT also) ; and 
2) Hashrates not always reported accurately;  On launch, sometimes Nanominer will report between 60 and 70 Mh/s for this card until you restart Nanominer (does this on 5700 XT also).  It can report this on one card, or the other, or both.  UPDATE: I discovered this happens when I "Run as Administrator" in Windows 10 Pro.  When I change the desktop launch icon to not do that, it reports properly... so this is not really a problem after all.
